### PR TITLE
added bcftools convert

### LINKED ISSUE #1903: new module: bcftools convert

### PR DESCRIPTION
**Pull request by nvnieuwk**
## PR checklist

Closes #1903 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/bcftools/convert/main.nf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::bcftools=1.15.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bcftools:1.15.1--h0ea216a_0':
+        'quay.io/biocontainers/bcftools:1.15.1--h0ea216a_0' }"
+
+    input:
+    tuple val(meta), path(input), path(input_index), path(bed)
```
Comment (matthdsm) [inline modules/bcftools/convert/main.nf:11]: Most `bcftools` modules use `vcf` and `tbi` as input name, perhaps it makes sense to keep this naming scheme?
Also, I'd pull the bed file from the input tuple

File: modules/bcftools/convert/main.nf [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def regions = bed ? "--regions-file $bed" : ""
+    def reference = fasta ?  "--fasta-ref $fasta" : ""
+
+    """
+    bcftools convert \\
+        $regions \\
```
Comment (matthdsm) [inline modules/bcftools/convert/main.nf:30]:
```suggestion
        $args \\
        $regions \\
```
`$args` is missing here

File: modules/bcftools/convert/main.nf (hunk 1 above)
Comment (nvnieuwk) [reply modules/bcftools/convert/main.nf:11]: The `input` in this case isn't always a `vcf` or `tbi`, so maybe `input` is the better name?

File: modules/bcftools/convert/main.nf (hunk 2 above)
Comment (nvnieuwk) [reply modules/bcftools/convert/main.nf:30]: oh oops

File: modules/bcftools/convert/main.nf (hunk 1 above)
Comment (matthdsm) [reply modules/bcftools/convert/main.nf:11]: good call!

Comment (nvnieuwk) [reply modules/bcftools/convert/main.nf:11]: possible other inputs are `tsv`, `hap`, `sample`...

File: modules/bcftools/convert/main.nf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def regions = bed ? "--regions-file $bed" : ""
+    def reference = fasta ?  "--fasta-ref $fasta" : ""
+
+    """
+    bcftools convert \\
+        $args \\
+        $regions \\
+        --output ${prefix}.vcf.gz \\
```
Comment (matthdsm) [inline modules/bcftools/convert/main.nf:32]: Output format is set with option `--output-type`, you can't be sure it'll be `vcf.gz`

Comment (nvnieuwk) [reply modules/bcftools/convert/main.nf:32]: good point! I'll look into it!

Comment (matthdsm) [reply modules/bcftools/convert/main.nf:32]: although, that seems to be the consensus among other modules.

File: modules/bcftools/convert/main.nf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::bcftools=1.15.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bcftools:1.15.1--h0ea216a_0':
+        'quay.io/biocontainers/bcftools:1.15.1--h0ea216a_0' }"
+
+    input:
+    tuple val(meta), path(input), path(input_index), path(bed)
```
Comment (matthdsm) [inline modules/bcftools/convert/main.nf:11]:
```suggestion
    tuple val(meta), path(input), path(input_index)
    path(bed)
```

**Review by matthdsm — APPROVED** [review]
👍🏻

**Pull request merged**